### PR TITLE
Unique Binary Tree and Catalan Number Issue #1436

### PR DESCRIPTION
**Pull request by charlie219**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

YES

### Description
As assigned by  @iamrajiv in issue #1436, I have made the following addition to the repo.
- Number of different Binary trees can be formed with N distinct nodes. (DP)
- Catalan Number. (Maths)

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

#1436

### CODE REVIEW

**Comment by charlie219** [comment]
@atarax665 @plazzy99 please review the PR

File: Python/README.md [hunk 1]
```diff
@@ -228,6 +229,8 @@
 - [Coin Change Problem](dp/CoinChange.py)
 - [Longest Common Subsequence](dp/Longest_Common_Subsequence.py)
 - [Longest Increasing Subsequence](dp/Longest_Increasing_Subsequence.py)
+- [Binary Trees with N nodes](dp/binary_trees_with_n_nodes.py)
```
Comment (HarshCasper) [inline Python/README.md:232]: Don't push more than one logical code in a single PR. You can always create corresponding PRs for all the different implementations around one algorithm.

File: Python/dp/Unique_BST.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+"""
+
+# Catalan_Number (N) = ((2*N)!) / ((N+1)!*N!)
+
+# Global Variables
+MOD = 10**9+7
+facto = [1]  # Factorial Table
+
+# To construct the factorial numbers using DP
+
```
Comment (HarshCasper) [inline Python/dp/Unique_BST.py]: Remove Extra Lines of Code

**Review by HarshCasper — CHANGES_REQUESTED** [review]
The Code looks absolute clean and well understanding! 

Great Work. Just a few changes requested and it will be ready to merge.

File: Python/README.md (hunk 1 above)
Comment (charlie219) [reply Python/README.md:232]: The pushed algorithms have similar approach i.e Catalan Number, hence I added those Algorithms too.
I will surely delete the Unique_BST.py and make another PR for this.

**Comment by charlie219** [comment]
@HarshCasper , the requested changes are made. I have deleted the Unique_BST.py which was assigned to me in Issue #1455 . I will make a separate PR for that issue.

File: Python/README.md [hunk 3]
```diff
@@ -206,6 +206,7 @@
 - [Fibonacci number](math/Fibonacci.py)
 - [Strassen's Matrix Multiplication](math/Strassen's_Matrix_Mul.py)
 - [Prime Number Iterative and Recursive](math/prime_number.py)
+- [Cataln Number](math/Catalan_Number.py)
```
Comment (iamrajiv) [inline Python/README.md]: @charlie219 Fix the typo here.

**Review by iamrajiv — CHANGES_REQUESTED** [review]

File: Python/README.md (hunk 3 above)
Comment (charlie219) [reply Python/README.md]: Typo Corrected

**Review by vatsalkesarwani12 — APPROVED** [review]
👍🏻

**Review by iamrajiv — APPROVED** [review]

**Pull request merged**